### PR TITLE
EVG-5314: Expose image ID in rest v2 host API endpoint

### PR DESCRIPTION
**Pull request by julianedwards**
Added some testing here as well.

### CODE REVIEW

File: model/distro/distro.go [hunk 1]
```diff
@@ -119,6 +119,34 @@ func (d *Distro) IsParent(s *evergreen.Settings) bool {
...
+		return s
+	case evergreen.ProviderNameDockerMock:
+		s, _ := (*d.ProviderSettings)["image_url"].(string)
+		return s
+	case evergreen.ProviderNameGce:
+		s, _ := (*d.ProviderSettings)["image_name"].(string)
+		return s
+	case evergreen.ProviderNameVsphere:
+		s, _ := (*d.ProviderSettings)["template"].(string)
+		return s
```
Comment (tychoish) [inline model/distro/distro.go]: we shouldn't ignore the ok check.

Comment (julianedwards) [reply model/distro/distro.go]: If ok is false should we return an empty string or an error?

File: rest/model/host.go [hunk 2]
```diff
@@ -87,9 +88,14 @@ func (apiHost *APIHost) buildFromHostStruct(h interface{}) error {
 	apiHost.Status = ToAPIString(v.Status)
 	apiHost.UserHost = v.UserHost
 
+	imageId, err := v.Distro.GetImageID()
+	if err != nil {
+		return fmt.Errorf("problem getting image ID")
```
Comment (tychoish) [inline rest/model/host.go]: should wrap the error?

Comment (julianedwards) [reply rest/model/host.go]: I just followed the trend of this file and used `fmt.Errorf` (errors isn't even imported..). Can add import and wrap if you would like.

Comment (tychoish) [reply rest/model/host.go]: should always just import things if we need it. fmt.Errorf isn't a thing we're calling very much/at all (it doesn't annotate the error with stack, and in this case it's not a format string, so you'd want to do errors.New anyway (if there weren't an error you wanted to wrap.)

File: model/distro/distro.go [hunk 3]
```diff
@@ -119,6 +119,34 @@ func (d *Distro) IsParent(s *evergreen.Settings) bool {
...
+	case evergreen.ProviderNameDocker:
+		i = (*d.ProviderSettings)["image_url"]
+	case evergreen.ProviderNameDockerMock:
+		i = (*d.ProviderSettings)["image_url"]
+	case evergreen.ProviderNameGce:
+		i = (*d.ProviderSettings)["image_name"]
+	case evergreen.ProviderNameVsphere:
+		i = (*d.ProviderSettings)["template"]
+	default:
+		return "", nil
```
Comment (tychoish) [inline model/distro/distro.go]: should this be an error? or a thing we log?

Comment (julianedwards) [reply model/distro/distro.go]: There are just some providers that do not have an image id, for example, the `Mock` and `Static`. I can check for those two (and leave the empty string as the image id), and if the Provider is unknown then throw an error. How does that sound?

Comment (julianedwards) [reply model/distro/distro.go]: Done.

**Pull request merged**